### DISCUSSION TITLE
How to embed an icon in an option for a FluentSelect programmatically?

### DISCUSSION THREAD

**Discussion by AndyEdmonds** [microsoft/fluentui-blazor#324] [Q&A] [answered]
The examples for The FluentSelect at https://brave-cliff-0c0c93310.azurestaticapps.net/Select show that you can add an icon to an option declaratively:
``
<FluentSelect ValueChanged="@(e => selectValue2 = e)" TOption="string">
    <FluentOption Value="1">
        Search
        <FluentIcon Name="@FluentIcons.Search" Slot="start" Size="IconSize.Size20" Color="Color.Neutral" />
     </FluentOption>
``

Programmatically we are stuck with using just Option<TOPTION>. This has no way to embed an icon.
If you try creating a list of FluentOption classes warnings are created about accessing Value etc outside of the component.

How should this be done?

**Comment by vnbaaij** [comment, marked as answer]
Good question. I don't think there is a way to do this yet. Creating a list of `<FluentOption>`s is not the right way. The warnings you get then are to be expected.

What I could do is add a new `Icon` property to the `Option<>` class. The type of the property would be a `Tuple` with the needed 
icon parameters, i.e.:

```
   public (string Name, IconSize? Size, IconVariant? Variant, Color? Color, string? Slot) Icon { get; set; }
```

At the place you are generating your `Option` collection you would then do something like this to add an option with an icon.: 

```
// The 'null' values instruct the code to use the FluentIcon's default values for those parameters
{ new Option<string> { Value = "9", Text = "Nine", Icon = (@FluentIcons.AppsList,null,null, Color.Neutral, "start" ) } },
```

It would then get rendered like this:
<img width="218" alt="image" src="https://user-images.githubusercontent.com/1761079/219752426-1151b969-1d5e-48c8-9dc7-4e28d7fe2cad.png">

Makes sense?

**Reply by adeyinkaroyal** [reply]
Hi @vnbaaij, this would be awesome.

**Reply by vnbaaij** [reply]
The functionality I described is already in. See https://www.fluentui-blazor.net/Select#fromalistofoptiontit
